notebooks/population/grid_pop_experiment.py

```diff
@@ -0,0 +1,1218 @@
+
+
+def clip_sum_overlap(geometry: Polygon, rst: DataArray) -> float:
+    """Clip raster data to geometry and estimate population.
+
+    Method includes all cells that touch the provided geometry. Then scales
+    the cell value based on the fraction of overlap between the cell and the
+    geometry. The total is then then combined sum of these fractional cell
+    values.
+
+    Parameters
+    ----------
+    geometry : Polygon
+        Polygon representing the bounds to clip to. Must be in the same CRS are
+        the raster data.
+    rst : DataArray
+        Raster data array
+
+    Returns
+    -------
+    float
+        Sum values within clip, excluding nan values, and adjusting for
+        fractional overlap of cells/geometry.
+
+    """
+    # clip to geometry and include all cells that touch
+    try:
+        rst_clipped = rst.rio.clip([geometry], all_touched=True)
+    except NoDataInBounds:
+        return 0
+
+    # vectorise and drop masked/null cells (nodata regions)
+    try:
+        clipped_gdf = vectorize(rst_clipped.squeeze().astype(np.float32))
+    except ValueError:
+        # TODO: use this approach, should work for all conditions
+        clipped_gdf = vectorize(rst_clipped.squeeze(axis=0).astype(np.float32))
+    clipped_gdf.dropna(subset=["population"], inplace=True)
+
+    # calculate the fractional overlap within the OA
+    clipped_gdf["overlap"] = (
+        clipped_gdf.intersection(geometry).area / clipped_gdf.area
+    )
+
+    # calculate the OA estimate by scaling the cell population by the
+    # fractional overlap
+    oa_estimate = (clipped_gdf["overlap"] * clipped_gdf["population"]).sum()
+
+    return oa_estimate
+
+
+# %%
+# apply the clip_sum_overlap func to all OAs and calculate the difference
+oa_gdf["ghs"] = oa_gdf.geometry.apply(clip_sum_overlap, args=[xds_sc])
+oa_gdf["pop_diff"] = oa_gdf["mye"] - oa_gdf["ghs"]
+
+# %%
+# repeat above, but exclude cells below a threshold
+threshs = [1, 3, 5, 7, 10]
+for thresh in threshs:
+    logger.info(f"Calculating GHS-POP estimate using {thresh}...")
+    xds_thresh = xds_sc.copy()
+    xds_thresh = xds_thresh.where(xds_thresh > thresh)
+    oa_gdf[f"ghs_thresh_{thresh}"] = oa_gdf.geometry.apply(
+        clip_sum_overlap, args=[xds_thresh]
+    )
+    oa_gdf[f"pop_diff_thresh_{thresh}"] = (
+        oa_gdf["mye"] - oa_gdf[f"ghs_thresh_{thresh}"]
+    )
+
+# %%
+# build an interactive folium map displaying OA pop estimate difference
+m = oa_gdf.explore(
+    "pop_diff",
+    legend=True,
+    cmap="RdBu",
+    vmin=-600,
+    vmax=600,
+    tiles="https://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}{r}.png",
+    attr=(
+        '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMa'
+        'p</a> contributors &copy; <a href="https://carto.com/attributions">CA'
+        "RTO</a>"
+    ),
+    legend_kwds={"backgroundcolor": "white"},
+)
+
+m.save(os.path.join(here(), "outputs", "mye", f"{AREA_OF_INTEREST}.html"))
```

Comment:
Notebook breaks here: path does not exist, needs to be created manually.

Reply:
Dir creation (if it doesn't already exists) has been added in commit 5a8da25.